sklearn/metrics/pairwise_fast.pyx

```diff
@@ -41,28 +41,64 @@ def _chi2_kernel_fast(floating[:, :] X,
 
 def _sparse_manhattan(floating[::1] X_data, int[:] X_indices, int[:] X_indptr,
                       floating[::1] Y_data, int[:] Y_indices, int[:] Y_indptr,
-                      np.npy_intp n_features, double[:, ::1] D):
+                      double[:, ::1] D):
     """Pairwise L1 distances for CSR matrices.
-
     Usage:
-
     >>> D = np.zeros(X.shape[0], Y.shape[0])
-    >>> sparse_manhattan(X.data, X.indices, X.indptr,
+    >>> _sparse_manhattan(X.data, X.indices, X.indptr,
     ...                  Y.data, Y.indices, Y.indptr,
-    ...                  X.shape[1], D)
+    ...                   D)
     """
-    cdef double[::1] row = np.empty(n_features)
-    cdef np.npy_intp ix, iy, j
+    cdef np.npy_intp px, py, i, j, ix, iy
+    cdef double d = 0.0
+
+    cdef int m = D.shape[0]
+    cdef int n = D.shape[1]
 
     with nogil:
-        for ix in range(D.shape[0]):
-            for iy in range(D.shape[1]):
-                # Simple strategy: densify current row of X, then subtract the
-                # corresponding row of Y.
-                memset(&row[0], 0, n_features * sizeof(double))
-                for j in range(X_indptr[ix], X_indptr[ix + 1]):
-                    row[X_indices[j]] = X_data[j]
-                for j in range(Y_indptr[iy], Y_indptr[iy + 1]):
-                    row[Y_indices[j]] -= Y_data[j]
+        for px in prange(m):
```

Comment:
Maybe add a few comments describing in English what this implementation does, as was done previously.